providers/base/bin/prime_offload_tester.py

```diff
@@ -214,6 +278,74 @@ def run_offload_cmd(
             self.logger.info(log_str)
             timeout = 20
```

Comment:
I don't really like this function. Couldn't we handle the timeout from Python instead of modifying the command string?
Take a look at the [timeout helper script](https://github.com/canonical/checkbox/blob/main/checkbox-support/checkbox_support/helpers/timeout.py), maybe you can use the timeout decorator, or the `run_with_timeout` function.
This way of handling the timeout is quite confusing and possibly hard to test/debug.